openedx/features/philu_courseware/views.py

```diff
@@ -65,3 +65,23 @@ def record_and_fetch_competency_assessment(request, chapter_id):
             }, status=status.HTTP_201_CREATED)
     else:
         return Response({'message': serializer.errors}, status=status.HTTP_400_BAD_REQUEST)
+
+
+@api_view(['POST'])
+@view_auth_classes(is_authenticated=True)
+@renderer_classes([JSONRenderer, BrowsableAPIRenderer])
+def revert_user_post_assessment_attempts(request, course_id):
+    user = request.user
+    problem_id = request.data.get('problem_id')
+    validated_problem_id = validate_problem_id(problem_id)
+    reverted_attempts_count = CompetencyAssessmentRecord.objects.revert_user_post_assessment_attempts(
+                                problem_id=validated_problem_id, user=user)
+    has_deleted_post_assessment_attempts = reverted_attempts_count > 0
+    if has_deleted_post_assessment_attempts:
+        revert_user_attempts_from_edx(course_id, user, validated_problem_id)
+        success_response = {
+            'message': _('Your attempts has been reverted successfully')
+        }
+        return Response(success_response, status=status.HTTP_200_OK)
```

Comment:
How about we send the response message here in the response instead of declaring `success_response` as it is not being used anywhere else. 
Also wouldn't it be cleaner to just send the message in the body instead of sending it inside 'message' key? Because thats the pattern being used in edX APIs.

Reply:
I have removed the message as on frontend we will not be showing this message. But yes status code would help to take the decision.